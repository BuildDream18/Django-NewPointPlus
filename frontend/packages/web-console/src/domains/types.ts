type User = Readonly<{
    id: string | null;
    name: string | null;
}>;

export { User };
