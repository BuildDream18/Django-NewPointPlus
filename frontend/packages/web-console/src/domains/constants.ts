/* eslint-disable */
// ステータスコード
export const STATUS_400 = 400;
export const STATUS_403 = 403;
export const STATUS_500 = 500;
export const STATUS_503 = 503;

// レスポンスタイプ
export const SUCCESS_RESPONSE = 'success';
export const FAILED_RESPONSE = 'error';

// エラーメッセージ
export const NOT_EXIST_ACCOUNT_ERROR_MESSAGE =
    '一致するアカウントが存在しません。';
export const OTHER_REASON_ACCOUNT_ERROR_MESSAGE =
    '何らかの理由でエラーが発生しました。カスタマーサポートへお問い合わせください。';

/* eslint-enable */
