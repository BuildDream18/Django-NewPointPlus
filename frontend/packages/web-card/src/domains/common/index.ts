import * as Api from '../../../../api/src/api/web-card';
import * as ApiModel from '../../../../api/src/api/web-card/model';
import store from '../../store';
import * as C from '../constants';

/**
 * アクセストークン失効APIをコールし、Storeの認証情報(auth/)を削除する
 * @param なし
 * @return なし
 */

export const expireAuthToken = async (): Promise<void> => {
    const refreshToken = store.getters.refreshToken;
    const authToken = store.getters.authToken;
    await Api.expireAuthTokenApi(refreshToken, authToken);
    return store.dispatch('clearAuthAction');
};

/**
 * アクセストークン更新APIをコールし、Storeの認証情報(auth/)を更新する
 * @param なし
 * @return なし
 */

export const updateAuthToken = async (): Promise<void> => {
    try {
        const refreshToken = store.getters.refreshToken;
        const { data } = await Api.updateAuthTokenApi({
            refresh_token: refreshToken,
        });
        if (data.type === C.SUCCESS_RESPONSE) {
            const payload = {
                authToken: data.auth_token,
                tokenExpirationDate: data.token_expiration_date,
                refreshToken: data.refresh_token,
            };
            return store.dispatch('setAuthAction', payload);
        } else {
            throw new Error();
        }
    } catch (err) {
        return Promise.reject();
    }
};
