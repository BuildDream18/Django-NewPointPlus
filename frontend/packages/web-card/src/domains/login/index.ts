import * as Api from '../../../../api/src/api/web-card';
import * as ApiModel from '../../../../api/src/api/web-card/model';
import store from '../../store';
import * as C from '../constants';

/**
 * サービス情報取得APIをコールし、Storeに企業情報を保存する
 * @param なし
 * @return なし
 */
type ErrorDetail = {
    status: number | undefined;
};

export const getServiceInfo = async (): Promise<void | ErrorDetail> => {
    try {
        const { data } = await Api.getServiceInfoApi();
        if (data.type === C.SUCCESS_RESPONSE) {
            const payload = {
                providerName: data.provider_name,
                serviceLogoUrl: data.service_logo_url,
                serviceName: data.service_name,
            };
            store.dispatch('setCorpInfoAction', { ...payload });
            return Promise.resolve();
        } else {
            return Promise.reject({ status: undefined });
        }
    } catch (errorRes) {
        const status = { status: errorRes.status };
        return Promise.reject(status);
    }
};

/**
 * カードアクセストークン発行APIをコールし、Storeにトークンを保存する
 * @param {string} cardNum - ログイン画面で入力したカード番号。
 * @param {string} pinNum - ログイン画面で入力したPIN番号。
 * @return なし
 */

export const issueAuthToken = async (
    cardNum: string,
    pinNum: string
): Promise<void | string> => {
    try {
        const requestParams: ApiModel.IssueAuthTokenApiParams = {
            card_number: cardNum,
            pin_number: pinNum,
        };
        const { data } = await Api.issueAuthTokenApi(requestParams);

        if (data.type === C.SUCCESS_RESPONSE) {
            const payload = {
                authToken: data.auth_token,
                tokenExpirationDate: data.token_expiration_date,
                refreshToken: data.refresh_token,
            };
            store.dispatch('setAuthAction', payload);
            return Promise.resolve();
        } else {
            throw new Error();
        }
    } catch (errorRes) {
        if (errorRes.status === C.STATUS_400) {
            return Promise.reject(C.NOT_EXIST_CARD_ERROR_MESSAGE);
        }
        return Promise.reject(C.OTHER_REASON_CARD_ERROR_MESSAGE);
    }
};
