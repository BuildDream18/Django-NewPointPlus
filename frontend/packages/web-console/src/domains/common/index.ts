import * as Api from '../../../../api/src/api/web-console';
import store from '../../store';
import * as C from '../constants';

/**
 * アクセストークン失効APIをコールし、Storeの認証情報(auth/)を削除する
 * @param なし
 * @return なし
 */

export const expireAuthToken = async (): Promise<void> => {
    const refreshToken = store.getters.refreshToken;
    const authToken = store.getters.authToken;
    await Api.expireAuthTokenApi(refreshToken, authToken).catch((e) => {
        console.log(e);
    });
    store.dispatch('clearAuthAction');
};

/**
 * アクセストークン更新APIをコールし、Storeの認証情報(auth/)を更新する
 * @param なし
 * @return なし
 */

export const updateAuthToken = async (): Promise<void> => {
    try {
        const refreshToken = store.getters.refreshToken;
        const { data } = await Api.updateAuthTokenApi({
            refresh_token: refreshToken,
        });
        if (data.type === C.SUCCESS_RESPONSE) {
            const payload = {
                authToken: data.token,
                tokenExpirationDate: data.token_expiration_date,
                refreshToken: data.refresh_token,
            };
            return store.dispatch('setAuthAction', payload);
        } else {
            throw new Error();
        }
    } catch (err) {
        return Promise.reject();
    }
};
