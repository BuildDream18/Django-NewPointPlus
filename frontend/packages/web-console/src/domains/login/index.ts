import * as Api from '../../../../api/src/api/web-console';
import * as ApiModel from '../../../../api/src/api/web-console/model';
import store from '../../store';
import * as C from '../../domains/constants';

/**
 * コンソールアクセストークン発行APIをコールし、Storeにトークンを保存する
 * @param {string} email - ログイン画面で入力したメール。
 * @param {string} loginPassword - ログイン画面で入力したログインパスワード。
 * @param {string} sendEmailFlag - ログイン画面で入力したメールフラグを送信する。
 * @return なし
 */

export const issueAuthToken = async (
    email: string,
    loginPassword: string,
    sendEmailFlag: number
): Promise<void | string> => {
    try {
        const requestParams: ApiModel.IssueAuthTokenApiParams = {
            email: email,
            login_password: loginPassword,
            send_email_flag: sendEmailFlag,
        };
        const { data } = await Api.issueAuthTokenApi(requestParams);

        // check login success
        if (!data.status) {
            if (data.type === C.SUCCESS_RESPONSE) {
                const payload = {
                    authToken: data.token,
                    tokenExpirationDate: data.token_expiration_date,
                    refreshToken: data.refresh_token,
                };
                store.dispatch('setAuthAction', payload);
                return Promise.resolve();
            }
        }
        // check login and else
        else if (data.status) {
            return Promise.reject(data.status.message);
        } else {
            throw new Error();
        }
    } catch (errorRes) {
        if (errorRes.data.status) {
            return Promise.reject(errorRes.data.status.message);
        }
        if (errorRes.status === C.STATUS_400) {
            return Promise.reject(C.NOT_EXIST_ACCOUNT_ERROR_MESSAGE);
        }
        return Promise.reject(C.OTHER_REASON_ACCOUNT_ERROR_MESSAGE);
    }
};
