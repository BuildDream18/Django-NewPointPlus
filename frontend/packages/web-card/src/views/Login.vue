<template>
    <v-container class="pa-0">
        <v-row
            no-gutters
            justify="center"
            style="height: 100vh"
            align-content="center"
        >
            <v-col>
                <v-layout justify-center>
                    <v-avatar
                        v-if="isExistCorpImage"
                        color="white"
                        size="120"
                        class="text-center"
                        justify="center"
                    >
                        <img :src="serviceLogoUrl" />
                    </v-avatar>
                    <div v-else>{{ serviceName }}</div>
                </v-layout>
            </v-col>
            <v-col cols="12" class="pa-0 mt-2">
                <div class="text-center">{{ providerName }}</div>
            </v-col>
            <v-col cols="12" md="6" class="mt-8 px-4 py-0">
                <v-form>
                    <v-text-field
                        v-model="form.card.number"
                        placeholder="カード番号"
                        outlined
                        class="mb-4"
                        hide-details="auto"
                        :rules="[]"
                        :error="form.card.error"
                    ></v-text-field>
                    <v-text-field
                        v-model="form.pin.number"
                        placeholder="PIN番号"
                        outlined
                        height="56"
                        class="mb-7"
                        hide-details="auto"
                        :append-icon="
                            form.pin.isMask ? 'visibility' : 'visibility_off'
                        "
                        :type="form.pin.isMask ? 'password' : 'text'"
                        :error="form.pin.error"
                        :error-messages="form.errorMessages"
                        @click:append="form.pin.isMask = !form.pin.isMask"
                    ></v-text-field>
                    <v-btn
                        color="primary"
                        class="white--text font-weight-bold py-4"
                        height="56"
                        block
                        elevation="0"
                        :disabled="form.isInvalid"
                        @click="handleLogin"
                    >
                        ログイン
                    </v-btn>
                </v-form>
            </v-col>
        </v-row>
    </v-container>
</template>
<script lang="ts">
import { getServiceInfo, issueAuthToken } from '../domains/login';
import { Component, Vue, Watch } from 'vue-property-decorator';
import { mapGetters } from 'vuex';

@Component({
    computed: {
        ...mapGetters(['providerName', 'serviceLogoUrl', 'serviceName']),
    },
})
export default class Login extends Vue {
    protected providerName!: string;
    protected serviceLogoUrl!: string;
    protected serviceName!: string;
    form = {
        card: {
            number: '',
            error: false,
        },
        pin: {
            number: '',
            error: false,
            isMask: true,
        },
        isInvalid: true,
        errorMessages: '',
    };
    get formStatus(): string[] {
        return [this.form.card.number, this.form.pin.number];
    }
    get isExistCorpImage(): boolean {
        return Boolean(this.serviceLogoUrl);
    }
    @Watch('formStatus')
    isFilled(): void {
        this.form.card.number && this.form.pin.number
            ? (this.form.isInvalid = false)
            : (this.form.isInvalid = true);
    }
    created(): void {
        this.callGetServiceInfo();
    }
    callGetServiceInfo(): void {
        getServiceInfo().catch((err) => {
            if (err && err.status) {
                if (err.status < 500) {
                    this.$router.push({ name: 'ClientErrorPage' });
                } else {
                    this.$router.push({ name: 'ServerErrorPage' });
                }
            }
        });
    }
    handleLogin(): void {
        issueAuthToken(this.form.card.number, this.form.pin.number)
            .then(() => this.$router.push({ name: 'Top' }))
            .catch((errMsg) => {
                this.form.errorMessages = errMsg;
            });
    }
}
</script>
