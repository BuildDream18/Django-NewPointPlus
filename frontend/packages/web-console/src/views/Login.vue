<template>
    <v-app>
        <v-app-bar
            app
            flat
            elevation="1"
            color="transparent"
            class="d-flex justify-center"
        >
            <div class="d-flex align-center">
                <v-img src="~@/assets/cmn/logo.svg" width="132"></v-img>
            </div>
        </v-app-bar>
        <v-container fill-height>
            <v-card max-width="364" class="ma-auto pa-8 pb-12" outlined>
                <v-card-text class="pa-0">
                    <div class="ma-0 pt-1 pb-2 px-0">
                        <v-text-field
                            v-model="user.email"
                            :error="isError"
                            :rules="rules.email"
                            type="text"
                            :label="$t('user.emailAddress')"
                            hide-details="auto"
                            outlined
                            dense
                            maxlength="256"
                        ></v-text-field>
                    </div>

                    <div class="ma-0 pt-2 pb-2 px-0">
                        <v-text-field
                            v-model="user.login_password"
                            :error="isError"
                            :rules="rules.password"
                            :type="isPasswordDisplay ? 'text' : 'password'"
                            hide-details="auto"
                            dense
                            outlined
                            maxlength="100"
                            :label="$t('user.password')"
                            :append-icon="
                                isPasswordDisplay
                                    ? 'visibility_off'
                                    : 'remove_red_eye'
                            "
                            @click:append="
                                isPasswordDisplay = !isPasswordDisplay
                            "
                        ></v-text-field>
                    </div>

                    <span class="red--text" style="white-space: pre-line">
                        {{ message }}
                    </span>
                </v-card-text>
                <v-btn
                    id="login-btn"
                    depressed
                    block
                    large
                    class="mt-5"
                    color="primary"
                    :disabled="!canClick"
                    @click="handleLogin"
                >
                    {{ $t('user.login') }}
                </v-btn>
            </v-card>
        </v-container>
        <layout-footer></layout-footer>
    </v-app>
</template>

<script lang="ts">
import { Vue, Component, Watch } from 'vue-property-decorator';
import LayoutFooter from '../components/Footer.vue';
import { issueAuthToken } from '../domains/login';
import { mapGetters } from 'vuex';

@Component({
    components: { LayoutFooter },
    computed: {
        ...mapGetters(['authToken']),
    },
})
export default class Login extends Vue {
    valid = false;
    isPasswordDisplay = false;
    isError = false;
    isNoInput = false;
    canClick = false;
    private rules = {
        email: [
            (v: string) => !!v || '',
            (v: string) => {
                const pattern =
                    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
                return pattern.test(v) || 'メールアドレスが正しくありません。';
            },
        ],
        password: [(v: string) => !!v || ''],
    };

    private user = {
        email: '',
        login_password: '',
        send_email_flag: 1,
    };
    private message = '';
    private redirectFrom: { redirect?: string } = {};
    private authToken!: string;

    created(): void {
        if (this.authToken) {
            this.$router.push({ name: 'History' });
        }
    }

    @Watch('user.email')
    onUserIdChanged(email: string): void {
        this.user.email = email;
        this.changeLoginButtonState();
    }

    @Watch('user.login_password')
    onPasswordChanged(loginPassword: string): void {
        this.user.login_password = loginPassword;
        this.changeLoginButtonState();
    }

    handleLogin(): void {
        issueAuthToken(
            this.user.email,
            this.user.login_password,
            this.user.send_email_flag
        )
            .then(() => {
                this.redirectFrom = this.$route.query;
                if (
                    this.redirectFrom.redirect &&
                    this.redirectFrom.redirect !== '/'
                ) {
                    this.$router.push(this.redirectFrom.redirect);
                } else {
                    this.$router.push({ name: 'History' });
                }
            })
            .catch((errMsg) => {
                this.message = errMsg;
            });
    }

    changeLoginButtonState(): void {
        this.canClick =
            this.isTextInputed(this.user.email) &&
            this.isTextInputed(this.user.login_password);
    }

    isTextInputed(text: string): boolean {
        return text.length > 0 ? true : false;
    }
}
</script>
