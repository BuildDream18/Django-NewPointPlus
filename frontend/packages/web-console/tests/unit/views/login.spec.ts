import { createLocalVue, shallowMount, config } from '@vue/test-utils';
import Vuex from 'vuex';
import Login from '../../../src/views/Login.vue';
import { issueAuthToken } from '../../../src/domains/login';
import authStore from '../../../src/store/modules/auth';
import Vuetify from 'vuetify';

jest.mock('../../../src/domains/login');
const localVue = createLocalVue();
config.showDeprecationWarnings = false;
localVue.config.silent = true;
localVue.use(Vuex);
localVue.use(Vuetify);
const issueAuthTokenMock = issueAuthToken as jest.MockedFunction<
    typeof issueAuthToken
>;

describe('views/login', () => {
    let store: any;
    const authInitValue = {
        authenticationToken: '',
    };
    beforeEach(() => {
        issueAuthTokenMock.mockClear();
        store = new Vuex.Store({
            modules: {
                auth: {
                    state: authStore.state,
                    mutations: authStore.mutations,
                    actions: authStore.actions,
                    getters: authStore.getters,
                },
            },
        });
        store.dispatch('setAuthAction', authInitValue);
    });

    const $t = () => {
        // i18n
    };
    it('mount Login', async () => {
        issueAuthTokenMock.mockReturnValue(Promise.resolve());
        const wrapper = shallowMount(Login, {
            store,
            localVue,
            mocks: {
                $router: {
                    push: jest.fn(),
                },
                $route: {
                    path: '/',
                },
                $t,
            },
        });
        wrapper.setData({
            user: {
                email: 'aaa@gmail.com',
                login_password: '123456',
                send_email_flag: 1,
            },
            canClick: true,
        });
        const btn = wrapper.find('#login-btn');
        await btn.trigger('click');
        expect(wrapper.element).toMatchSnapshot();
    });
});
