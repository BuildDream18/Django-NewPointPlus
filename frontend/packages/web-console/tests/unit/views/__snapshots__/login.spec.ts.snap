// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/login mount Login 1`] = `
<v-app-stub
  id="app"
>
  <v-app-bar-stub
    app="true"
    class="d-flex justify-center"
    color="transparent"
    elevation="1"
    extensionheight="48"
    flat="true"
    src=""
    tag="header"
    value="true"
  >
    <div
      class="d-flex align-center"
    >
      <v-img-stub
        options="[object Object]"
        position="center center"
        src="~@/assets/cmn/logo.svg"
        transition="fade-transition"
        width="132"
      />
    </div>
  </v-app-bar-stub>
   
  <v-container-stub
    fill-height=""
    tag="div"
  >
    <v-card-stub
      class="ma-auto pa-8 pb-12"
      loaderheight="4"
      maxwidth="364"
      outlined="true"
      tag="div"
    >
      <v-card-text-stub
        class="pa-0"
      >
        <div
          class="ma-0 pt-1 pb-2 px-0"
        >
          <v-text-field-stub
            backgroundcolor=""
            clearicon="$clear"
            dense="true"
            errorcount="1"
            errormessages=""
            hidedetails="auto"
            loaderheight="2"
            maxlength="256"
            messages=""
            outlined="true"
            rules="[Function],[Function]"
            successmessages=""
            type="text"
            value="aaa@gmail.com"
          />
        </div>
         
        <div
          class="ma-0 pt-2 pb-2 px-0"
        >
          <v-text-field-stub
            appendicon="remove_red_eye"
            backgroundcolor=""
            clearicon="$clear"
            dense="true"
            errorcount="1"
            errormessages=""
            hidedetails="auto"
            loaderheight="2"
            maxlength="100"
            messages=""
            outlined="true"
            rules="[Function]"
            successmessages=""
            type="password"
            value="123456"
          />
        </div>
         
        <span
          class="red--text"
          style="white-space: pre-line;"
        >
          
                    
                
        </span>
      </v-card-text-stub>
       
      <v-btn-stub
        activeclass=""
        block="true"
        class="mt-5"
        color="primary"
        depressed="true"
        id="login-btn"
        large="true"
        tag="button"
        type="button"
      >
        
                
            
      </v-btn-stub>
    </v-card-stub>
  </v-container-stub>
   
  <layout-footer-stub />
</v-app-stub>
`;
